### PR TITLE
[Merged by Bors] - #CU-2mezh63 junod client for cosmos for xcvm

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**

### CODE REVIEW

**Review by cor — APPROVED** [review]
LGTM

File: flake.nix [hunk 1]
```diff
@@ -458,61 +478,85 @@
...
-                rnix-lsp
-                rust-nightly
-                taplo
-                wasm-optimizer
-                xorriso
-                zlib.out
-              ];
-              NIX_PATH = "nixpkgs=${pkgs.path}";
+
+            junod = pkgs.callPackage ./.nix/junod.nix { };
```
Comment (cor) [inline flake.nix:508]: I think you accidentally added this as a devshell instead of as a package

**Review by cor — DISMISSED** [review]
one fix needed

**Review by hussein-aitlahcen — APPROVED** [review]

File: flake.nix [hunk 2]
```diff
@@ -458,61 +478,85 @@
...
               buildInputs = with packages; [ rust-nightly ];
               NIX_PATH = "nixpkgs=${pkgs.path}";
             });
 
+            developers-xcvm = developers-minimal.overrideAttrs (base: {
+              buildInputs = with packages;
+                base.buildInputs ++
+                [
+                  junod
+                  devnet-dali
```
Comment (cor) [inline flake.nix:548]: I'm not sure if we need to include junod and devnet-dali in the developer shells. I think these will only be used by other `nix run ...` commands

File: flake.nix [hunk 3]
```diff
@@ -120,45 +131,53 @@
...
+                  let
+                    customFilter = name: type:
+                      (
+                        !(type == "directory"
+                        && builtins.elem (baseNameOf name) dir-blacklist)
+                      )
+                      && (
+                        !(type == "file"
+                        && builtins.elem (baseNameOf name) file-blacklist)
+                      );
```
Comment (cor) [inline flake.nix:174]: Please change to
```
!(type == directory) && builtins.elem (baseNameOf name) (dir-blacklist ++ file-blacklist)
```

**Review by cor — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (dzmitry-lahoda) [reply flake.nix:508]: bad merge

File: flake.nix (hunk 2 above)
Comment (dzmitry-lahoda) [reply flake.nix:548]: i will use junod for sure. 

why you think that only nix run will use it?

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Pull request closed without merging**